applications/GeoMechanicsApplication/custom_elements/transient_thermal_element.h

```diff
@@ -0,0 +1,303 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Mohamed Nabi
+//                   John van Esch
+//                   Gennady Markelov
+//
+
+#pragma once
+
+#include "includes/element.h"
+#include "geo_mechanics_application_variables.h"
+#include "custom_constitutive/thermal_dispersion_law.h"
+#include "includes/serializer.h"
+
+namespace Kratos {
+
+template <unsigned int TDim, unsigned int TNumNodes>
+class KRATOS_API(GEO_MECHANICS_APPLICATION) TransientThermalElement : public Element {
+public:
+    KRATOS_CLASS_INTRUSIVE_POINTER_DEFINITION(TransientThermalElement);
+
+    explicit TransientThermalElement(IndexType NewId = 0) : Element(NewId) {}
+
+    TransientThermalElement(IndexType             NewId,
+                            GeometryType::Pointer pGeometry)
+        : Element(NewId, pGeometry)
+    {
+    }
+
+    TransientThermalElement(IndexType               NewId,
+                            GeometryType::Pointer   pGeometry,
+                            PropertiesType::Pointer pProperties)
+        : Element(NewId, pGeometry, pProperties)
+    {
+    }
+
+    ~TransientThermalElement() override = default;
+    TransientThermalElement(const TransientThermalElement&) = delete;
+    TransientThermalElement& operator=(const TransientThermalElement&) = delete;
+
+    Element::Pointer Create(IndexType               NewId,
```

Comment:
I thought all ::Pointers were shared pointer types, is it different for elements?